nautobot_ssot/jobs/base.py

```diff
@@ -110,6 +110,18 @@ def sync_data(self):
         - self.sync       (Sync instance tracking this job execution)
         - self.job_result (as per Job API)
         """
+
+        def record_memory_trace(step: str):
+            """Helper function to record memory usage and reset tracemalloc stats."""
+            memory_size, memory_peak = tracemalloc.get_traced_memory()
+            setattr(self.sync, f"{step}_memory_size", memory_size)
+            setattr(self.sync, f"{step}_memory_peak", memory_peak)
+            self.sync.save()
+            self.log_info(
+                message=(f"Traced memory for {step} (Current, Peak): {memory_size} bytes, {memory_peak} bytes")
+            )
+            tracemalloc.clear_traces()
```

Comment:
I'm still struggling to understand the meaning/significance of these various stats, even after a quick skim of the `tracemalloc` docs. I gather that by using `clear_traces` at each stage, we're getting only incremental memory stats for that stage - so the total memory used by the overall sync process is at least (sum of all "current" measurements)? And for any given stage, the actual peak memory usage is (sum of all previous stages' "current" measurements plus the "peak" measurement for the current stage)?

I think at a minimum it would be good to have a paragraph or two in the README explaining how to read these statistics and what meaning to take from them.

Reply:
Your understanding is the same as mine, so for each step we capture the final memory allocation and the peak.
I will improve the README for this stats, for developers to understand what they mean